packages/wonder-blocks-popover/src/components/__tests__/popover.test.tsx

```diff
@@ -356,4 +356,191 @@ describe("Popover", () => {
+                    name: "Button inside popover",
+                }),
+            ).toHaveFocus();
+        });
+
+        it("should allow flowing focus correctly even if the popover remains open", async () => {
+            // Arrange
+            render(
+                <>
+                    <Button>Prev focusable element outside</Button>
+                    <Popover
+                        onClose={jest.fn()}
+                        content={
+                            <PopoverContent
+                                title="Popover title"
+                                content="content"
+                                actions={<Button>Button inside popover</Button>}
+                            />
+                        }
+                    >
+                        <Button>Open default popover</Button>
+                    </Popover>
+                    <Button>Next focusable element outside</Button>
+                </>,
+            );
+
+            // Focus on the first element outside the popover
+            userEvent.tab();
+            // open the popover by focusing on the trigger element
+            userEvent.tab();
+            userEvent.keyboard("{enter}");
+
+            // Wait for the popover to be open.
+            await screen.findByRole("dialog");
+
+            // Act
+            // Focus on the next element after the popover
+            userEvent.tab();
+
+            // Assert
+            expect(
+                screen.getByRole("button", {
+                    name: "Next focusable element outside",
+                }),
+            ).toHaveFocus();
+        });
+
+        it("should allow circular navigation when the popover is open", async () => {
+            // Arrange
+            render(
+                <>
+                    <Button>Prev focusable element outside</Button>
+                    <Popover
+                        onClose={jest.fn()}
+                        content={
+                            <PopoverContent
+                                title="Popover title"
+                                content="content"
+                                actions={<Button>Button inside popover</Button>}
+                            />
+                        }
+                    >
+                        <Button>Open default popover</Button>
+                    </Popover>
+                    <Button>Next focusable element outside</Button>
+                </>,
+            );
+
+            // Focus on the first element outside the popover
+            userEvent.tab();
+            // open the popover by focusing on the trigger element
+            userEvent.tab();
+            userEvent.keyboard("{enter}");
+
+            // Wait for the popover to be open.
+            await screen.findByRole("dialog");
+
+            // Focus on the next element after the popover
+            userEvent.tab();
+
+            // Focus on the document body
+            userEvent.tab();
```

Comment:
**suggestion (non-blocking):** it might good to assert what some of these elements are in addition to the comments.